src/aoai-simulated-api/src/aoai_simulated_api/generator/openai.py

```diff
@@ -494,79 +495,120 @@ def create_chat_completion_response(
+            yield "\n"
+
+            chunk_string = json.dumps(
+                {
+                    "choices": [
+                        {
+                            "content_filter_results": {},
+                            "delta": {"content": "", "role": "assistant"},
+                            "finish_reason": None,
+                            "index": 0,
+                        }
+                    ],
+                    "created": int(time.mktime(datetime.datetime.now(datetime.UTC).timetuple())),
+                    "id": response_id,
+                    "model": "gpt-35-turbo",
+                    "object": "chat.completion.chunk",
+                    "system_fingerprint": None,
+                },
+                separators=(",", ":"),
+            )
+            yield "data: " + chunk_string + "\n"
+            yield "\n"
+
+            # send content chunks
             space = ""
-            role = "assistant"
-            for word in generated_content.split(" "):
+            created_timestamp = int(time.time())
+            for word in words:
                 chunk_string = json.dumps(
                     {
-                        "id": "chatcmpl-" + nanoid.non_secure_generate(size=29),
-                        "object": "chat.completion.chunk",
-                        "created": int(time.time()),
-                        "model_name": model_name,
-                        "system_fingerprint": None,
                         "choices": [
                             {
-                                "delta": {
-                                    "content": space + word,
-                                    "function_call": None,
-                                    "role": role,
-                                    "tool_calls": None,
-                                    "finish_reason": None,
-                                    "index": 0,
-                                    "logprobs": None,
-                                    "content_filter_results": {
-                                        "hate": {"filtered": False, "severity": "safe"},
-                                        "self_harm": {"filtered": False, "severity": "safe"},
-                                        "sexual": {"filtered": False, "severity": "safe"},
-                                        "violence": {"filtered": False, "severity": "safe"},
-                                    },
+                                "content_filter_results": {
+                                    "hate": {"filtered": False, "severity": "safe"},
+                                    "self_harm": {"filtered": False, "severity": "safe"},
+                                    "sexual": {"filtered": False, "severity": "safe"},
+                                    "violence": {"filtered": False, "severity": "safe"},
                                 },
-                            },
+                                "delta": {"content": space + word},
```

Comment:
I was just wondering if you could simply add the `space` to the end (i.e. `word + space`) and avoid having to set it each time around the loop with `space = " "`. I guess the real API prepends spaces, right?

Reply:
Yeah the actual service prepends spaces